docs/content/latest/explore/ysql-language-features/partitions.md

```diff
@@ -0,0 +1,259 @@
+
+A partitioned table and its partitions have hierarchical structure and are subject to its rules, except the following:
+
+- If there are no partitions, you can use the `ONLY` clause to add or drop a constraint on the partitioned table. If there are partitions, you cannot use `ONLY` because adding or deleting constraints on the  partitioned tableis not supported in such cases.
+- Partitions cannot have columns which do not exist in their partitioned table. You can add a partition using `ALTER TABLE ... ATTACH PARTITION` if their columns match the partitioned table, including a `oid` column.
+- Partitions inherit `CHECK` and `NOT NULL` constraints of their partitioned table. This does not include `CHECK` constraints marked `NO INHERIT`.
+- The `NOT NULL` constraint cannot be added to a partition's column if the constraint is present in their partitioned table.
+
+Suppose you work with a database that includes the following table:
+
+```sql
+CREATE TABLE employees (
+  employee_no integer PRIMARY KEY,
+  name text,
+  department text,
+  change_date date
+);
+```
+
+*change_date* represents the date when any type of change occured on the employee record. This date is typically required when generating monthly reports and only the last month's data is queried. To reduce the amount of old data that  needs to be stored, You can reduce the amount of data to store and keep these data for one year, and on the first of every month you remove the data from the previous month. You can partition the `employees` table to meet these requirements. You start by specifying bounds corresponding to the partitioning method and partition key of the `employees` table. This means you create partitions as regular tables and YSQL generates partition constraints automatically based on the partition bound specification every time they have to be referenced.
+
+You can declare partioning on the `employees` table by first creating it as a partitioned table. You specify the `PARTITION BY` clause and supply it with the partitioning method such as `RANGE`  as well as a list of columns as a partition key, as shown in the following example:
+
+```sql
+CREATE TABLE employees (
+  employee_no integer PRIMARY KEY,
+  name text,
+  department text,
+  change_date date
+) 
+PARTITION BY RANGE (change_date);
+```
+
+You create the actual partitions as follows:
+
+```sql
+CREATE TABLE employees_2019_02 PARTITION OF employees
+  FOR VALUES FROM ('2019-02-01') TO ('2019-03-01');
+
+CREATE TABLE employees_2019_03 PARTITION OF employees
+  FOR VALUES FROM ('2019-03-01') TO ('2019-04-01');
+
+...
+CREATE TABLE employees_2020_11 PARTITION OF employees
+  FOR VALUES FROM ('2020-11-01') TO ('2020-12-01');
+
+CREATE TABLE employees_2020_12 PARTITION OF employees
```

Comment:
We do not support tablespaces yet, please change the queries below also.

Reply:
Removed tablespaces